pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -898,17 +898,29 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 	volumeToMount VolumeToMount,
 	actualStateOfWorld ActualStateOfWorldMounterUpdater) (volumetypes.GeneratedOperations, error) {
 
+	originalSpec := volumeToMount.VolumeSpec
+	// Translate to CSI spec if migration enabled
+	if useCSIPlugin(og.volumePluginMgr, volumeToMount.VolumeSpec) {
+		csiSpec, err := translateSpec(volumeToMount.VolumeSpec)
+		if err != nil {
+			return volumetypes.GeneratedOperations{}, volumeToMount.GenerateErrorDetailed("MapVolume.TranslateSpec failed", err)
+		}
+
+		volumeToMount.VolumeSpec = csiSpec
+	}
+
 	// Get block volume mapper plugin
-	var blockVolumeMapper volume.BlockVolumeMapper
 	blockVolumePlugin, err :=
 		og.volumePluginMgr.FindMapperPluginBySpec(volumeToMount.VolumeSpec)
```

Comment:
FindMapperPluginBySpec will not be able to operate when `volumeToMount.VolumeSpec = csiSpec` is set right?

Reply:
Given my understanding, csi plugin is [registered](https://github.com/kubernetes/kubernetes/blob/48def74e641f8dff50fcf7c33c2a412593e2fede/cmd/kubelet/app/plugins.go#L101) with kubelet, so`FindMapperPluginBySpec` should return the csi plugin? Otherwise, my manual test on raw block volume migration won't pass too.

Reply:
Ah .. you are right. I was thinking of the spec pointing to the specific CSI plugin but that is not what the Spec/`Find*PluginBySpec` looks for.